scripts/main.js

```diff
@@ -0,0 +1,190 @@
+// cSpell: ignore Apikey
+
+// ----------* object to track state of app *----------
+const state = {
+  image: undefined,
+  caption: undefined,
+  codeSnippet: undefined,
+  fetchedJson: undefined,
+};
```

Comment:
recommend setting these to `null` - it is generally not recommended to manually set things to undefined. Using null is more explicit. Also, in this case, there is no distinguishing between a key that you have added to your object, and one that isn't there, since both would be undefined and that is not ideal. (eg. if I called state.foo on this object, it would also be undefined)

Some discussions on the topic:

https://vvkchandra.medium.com/essential-javascript-mastering-null-vs-undefined-66f62c65d16b

https://stackoverflow.com/questions/6604749/what-reason-is-there-to-use-null-instead-of-undefined-in-javascript